webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/domain/reports/StatusReports.scala

```diff
@@ -343,23 +344,25 @@ final case class BlockStatusReport(
         }
         ComplianceLevel.compute(kept)
       // focus on a given sub-component name (can be present several time, or 0 which leads to N/A)
-      case FocusReport(component) => ComplianceLevel.sum(findChildren(component).map(_.compliance))
+      case FocusReport(component) =>
+        ComplianceLevel.sum(findChildren(component).map(_.compliance))
     }
   }
 
   def getValues(predicate: ComponentValueStatusReport => Boolean): Seq[ComponentValueStatusReport] = {
     subComponents.flatMap(_.getValues(predicate))
   }
 
-  def componentValues:                                                       List[ComponentValueStatusReport] = ComponentValueStatusReport.merge(getValues(_ => true))
-  def withFilteredElement(predicate: ComponentValueStatusReport => Boolean): Option[ComponentStatusReport]    = {
+  val componentValues: List[ComponentValueStatusReport] = ComponentValueStatusReport.merge(getValues(_ => true))
+
+  def withFilteredElement(predicate: ComponentValueStatusReport => Boolean): Option[ComponentStatusReport] = {
     subComponents.flatMap(_.withFilteredElement(predicate)) match {
       case Nil => None
       case l   => Some(this.copy(subComponents = l))
     }
   }
 
-  def status: ReportType = {
+  val status: ReportType = {
```

Comment:
I think we are oscillating between def and val because val (or lazy val) makes the number of allocation/memory consumption exploses. Did you look in history to see if we did the opposite change at some point?